pkg/plugins/builtin/script/script.go

```diff
@@ -0,0 +1,108 @@
+package script
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	gexec "os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ovh/utask/pkg/plugins/taskplugin"
+)
+
+// the script plugin execute scripts
+var (
+	Plugin = taskplugin.New("script", "0.1", exec,
+		taskplugin.WithConfig(validConfig, Config{}),
+	)
+)
+
+// Config is the configuration needed to execute a script
+type Config struct {
+	File    string   `json:"file,required"`
+	Argv    []string `json:"argv,omitempty"`
+	Timeout string   `json:"timeout_seconds,omitempty"`
+}
+
+func validConfig(config interface{}) error {
+	cfg := config.(*Config)
+
+	if cfg.File == "" {
+		return errors.New("file is missing")
+	}
+
+	if cfg.Timeout != "" {
+		if _, err := strconv.ParseUint(cfg.Timeout, 10, 64); err != nil {
+			return fmt.Errorf("timeout is wrong %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+func exec(stepName string, config interface{}, ctx interface{}) (interface{}, interface{}, error) {
+	cfg := config.(*Config)
+
+	var timeout time.Duration
+
+	if cfg.Timeout != "" {
+		t, _ := strconv.ParseInt(cfg.Timeout, 10, 64)
+		timeout = time.Duration(t)
```

Comment:
should we parse a duration string instead?
ie 5s, 1m, ...

Reply:
additionally, it's called "timeout_seconds" but is actually in ms

Reply:
I think it's more readable like this. WDYT @nomonamo ?
It's not in ms, you missed the conversion. I moved the conversion for more readability.

Reply:
Ok, I see the conversion down below. This is a lot _less_ readable in my opinion, since all the code paths manipulating a time.Duration before that seem to be buggy (ie, the comment I made).